app/src/main/java/org/fossasia/openevent/app/event/detail/EventDetailsPresenter.java

```diff
@@ -58,38 +60,43 @@ public void loadTickets(long eventId, boolean forceReload) {
                 });
     }
 
-    private void showEventInfo(Event event) {
-        if(eventDetailView == null)
-            return;
-
-        eventDetailView.showEventName(event.getName());
-
-        String[] startDate = event.getStartTime().split("T");
-        String[] endDate = event.getEndTime().split("T");
-
-        eventDetailView.showDates(startDate[0], endDate[0]);
-        eventDetailView.showTime(endDate[1]);
-    }
-
     private void processEventAndDisplay(Event event) {
         if(eventDetailView == null)
             return;
 
+        this.event = event;
+
         showEventInfo(event);
 
         List<Ticket> tickets = event.getTickets();
 
-        totalTickets = 0;
+        long totalTickets = 0;
         if(tickets != null) {
             for (Ticket thisTicket : tickets)
                 totalTickets += thisTicket.getQuantity();
         }
 
-        eventDetailView.showTicketStats(totalTickets > 0 ? totalAttendees : 0, totalTickets);
+        event.totalTickets.set(totalTickets);
+        event.totalAttendees.set(totalAttendees);
+        event.checkedIn.set(checkedInAttendees);
 
         eventDetailView.showProgressBar(false);
     }
 
+    private void showEventInfo(Event event) {
+        if(eventDetailView == null)
```

Comment:
There is a need. When the presenter is detached and a request takes time to return, it will cause memory leak. In short, the need of null check will always remain.